shared/src/lib.rs

```diff
@@ -499,7 +499,7 @@ pub struct ProcessingM2 {
     pub th_2: BytesHashLen,
     pub x: BytesP256ElemLen,
     pub g_y: BytesP256ElemLen,
-    pub plaintext_2: EdhocMessageBuffer,
+    pub plaintext_2: BufferPlaintext2,
```

Comment:
Checking my understanding: we go back to things like `BufferPlaintext2` so we can more easily slap a `EdhocBuffer<N>` on its type alias, whenever needed.

Reply:
BufferPlaintext2 is already an EdhocBuffer type alias; the trouble is that some users of this expected it to be an EdhocMessageBuffer and some to be a BufferPlaintext2, and if those are not aliases (and we probably want to customize the N at some point, esp. if we want to prove panic freedom), we'll need to be precise even if the aliases happen to point to the same type when the consts align.